src/core/packages/notifications/browser/src/types.ts

```diff
@@ -77,3 +76,37 @@ export interface IToasts {
   addDanger: (toastOrTitle: ToastInput, options?: any) => Toast;
   addError: (error: Error, options: ErrorToastOptions) => Toast;
 }
+
+/**
+ * Specifies the internal state of {@link NotificationCoordinatorPublicApi}.
+ */
+export interface NotificationCoordinatorState {
+  locked: boolean;
+  controller: string | null;
+}
+
+/**
+ * @public
+ * @description This is the public API for the notification coordinator.
+ * It allows to opt in to coordination of notifications and acquiring/releasing locks on the notifications displayed to the user.
+ */
+export interface NotificationCoordinatorPublicApi {
+  optInToCoordination: <
+    T extends Array<{
+      id: string;
+    }>
+  >(
+    $: Observable<T>,
+    cond: Parameters<typeof filter<NotificationCoordinatorState>>[0]
```

Comment:
Q: I tend to prefer 
```suggestion
    cond: (coordinatorState: NotificationCoordinatorState) => boolean
```

Or, if you want to provide all the possible parameters
```suggestion
    cond: (coordinatorState: NotificationCoordinatorState, index: number) => boolean
```

IMO, it's much easier to read and to understand what's expected. And, if the `filter` operator changes its signature, we only need to adapt the coordinator, not all users.

Reply:
Fair enough... I see your opt, I'll make changes